app/controllers/users_controller.rb

```diff
@@ -0,0 +1,36 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :ensure_admin!
+
+  def index
+    @users = User.all
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      redirect_to users_path
+      flash[:success] = "User updated"
```

Comment:
I doubled-checked and it does get evaluated.  If it's preferred have it before the redirect, I can move it.  Either way, I'll add some assertions for them later on today.  Thanks!